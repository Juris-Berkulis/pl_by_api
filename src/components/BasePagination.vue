<script setup lang="ts">
interface Props {
    pagesCount: number,
    changePage: (newPageNumber: number) => void,
    page: number,
};

const props = defineProps<Props>();

const isShowPageBtn = (pageNumber: number): boolean => {
    return pageNumber === 1 
        || pageNumber === props.pagesCount 
        || props.page === pageNumber 
        || pageNumber - props.page === 1 
        || props.page - pageNumber === 1
};

const isShowEllipsisBetweenPages = (pageNumber: number): boolean => {
    return pageNumber - props.page === 2 || props.page - pageNumber === 2
};
</script>

<template>
<div class="w-full flex justify-center gap-4">
    <template v-for="pageNumber of pagesCount" :key="pageNumber">
        <button 
            class="p-2"
            :class="[page === pageNumber ? 'bg-sky-300' : 'bg-sky-200']"
            v-if="isShowPageBtn(pageNumber)"
            @click="changePage(pageNumber)"
            :data-page="pageNumber"
        >{{ pageNumber }}</button>
        <span v-else-if="isShowEllipsisBetweenPages(pageNumber)">...</span>
    </template>
</div>
</template>

<style scoped>
</style>
