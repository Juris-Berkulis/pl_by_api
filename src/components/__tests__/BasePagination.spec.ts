import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import BasePagination from "../BasePagination.vue";

describe('BasePagination', () => {
    const pagesCount = 4;

    const wrapper = mount(BasePagination, {
        props: {
            pagesCount,
            changePage: () => {},
            page: 1,
        }
    });

    it('snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Contains text', () => {
        for (let page = 1; page <= pagesCount; page++) {
            expect(wrapper.text()).toContain(page);
        };
    });
});
