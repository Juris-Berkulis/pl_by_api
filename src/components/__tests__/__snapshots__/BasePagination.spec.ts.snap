// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BasePagination > snapshot 1`] = `"<div class=\\"w-full flex justify-center gap-4\\"><button class=\\"p-2 bg-sky-200 bg-sky-300\\">1</button><button class=\\"p-2 bg-sky-200\\">2</button><button class=\\"p-2 bg-sky-200\\">3</button><button class=\\"p-2 bg-sky-200\\">4</button></div>"`;
